src/rust/ide/src/controller/graph.rs

```diff
@@ -492,9 +507,20 @@ impl Handle {
     }
 
     /// Returns information about all the connections between graph's nodes.
-    pub fn connections(&self) -> FallibleResult<Connections> {
+    ///
+    /// Will use `self` as the context for span tree generation.
+    pub fn connections_unevaluated_ctx(&self) -> FallibleResult<Connections> {
+        let graph   = self.graph_info()?;
+        Ok(Connections::new(&graph,self))
+    }
+
+    /// Returns information about all the connections between graph's nodes.
+    pub fn connections
+    (&self, context:&impl SpanTreeContext) -> FallibleResult<Connections> {
         let graph = self.graph_info()?;
-        Ok(Connections::new(&graph))
+        // TODO perhaps this should merge given context with the metadata information
+        //      or perhaps this should just do exactly what it is told
+        Ok(Connections::new(&graph,context))
     }
```

Comment:
I would make functions:
* `connections(self)` - to provide connections with data from metadata
* `connections_enriched(self, context)` - to apply result of context before metadata - being used by executed graph.

Reply:
Leaving as-is, as discussed. For now that for consistency would require providing a number of mostly duplicated unused methods.
If ever needed, then they can provided or alternatively a design where graph controller holds context as a field should be reconsidered.